python/lsst/dax/distribution/DataGenServer.py

```diff
@@ -195,25 +190,37 @@ def __init__(self, cfg_file_name, min_chunk_num, max_chunk_num,
         assert 'chunker' in spec_globals, "Specification file must define a variable 'chunker'."
         chunker = spec_globals['chunker']
         all_chunks = chunker.getAllChunks()
-        self._chunks_to_send = {} # Dictionary of information on chunks to send
-        # Set of chunks to send, desirable to have in order but not essential.
-        self._chunks_to_send_set = set()
         print("Finding valid chunk numbers...")
-        for chunk in total_chunks:
-            if chunk in all_chunks:
-                chunk_info = ChunkInfo(chunk)
-                self._chunks_to_send[chunk] = chunk_info
-                self._chunks_to_send_set.add(chunk)
+        # Use provided information to build the set of chunks to generate.
+        chunk_logs_in.build(all_chunks)
+        # Use the input information/files to create the output logs.
+        self._chunk_logs = chunk_logs_in.createOutput(log_dir)
+        if log_dir is not None:
+            # Start logging
+            self._chunk_logs.write()
+        # Set of chunks to send
+        self._chunks_to_send_set = self._chunk_logs.result_set.copy()
         self._chunks_to_send_total = len(self._chunks_to_send_set)
-        self._limbo_count = 0 # number of chunks that had problems being created.
-        print("len(totalChunks)=", len(total_chunks),
-              "_chunks_to_send_total=", self._chunks_to_send_total)
+        self._limbo_count = 0  # number of chunks that had problems being created.
+        # Dictionary of information about chunks being sent.
+        # self._chunks_to_send only includes information about this run.
+        # self._chunk_logs may include information from previous runs.
+        self._chunks_to_send = {}
+        for chunk in self._chunks_to_send_set:
+            chunk_info = ChunkInfo(chunk)
+            self._chunks_to_send[chunk] = chunk_info
+        print("_chunks_to_send_total=", self._chunks_to_send_total)
 
         # Track all client connections so it is possible to
         # determine when the server's job is finished.
         self._active_client_count = 0
         self._active_client_mtx = threading.Lock()
 
+    def chunksToSendTotal(self):
+        """Return the total number of chunks to send.
+        """
+        return self._chunks_to_send_total
```

Comment:
Why not make _chunks_to_send_total public variable instead of adding this accessor method?

Reply:
I want it to be impossible for some other class to change its value.